core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/public/datasourceEditorDialog.css

```diff
@@ -325,15 +371,59 @@ aggregation-dialog-checkbox-label {
   height: 463px !important;
 }
 
+:where(.responsive)#modeler_dialog .model-deck-panel {
+  height: 100% !important;
+}
+
+:where(.responsive#modeler_dialog) .pentaho-tabBar {
+  background-size: contain;
+}
+:where(.responsive#modeler_dialog) :is(.pentaho-tabBar, .pentaho-tabBar .pentaho-tabWidget) {
+  height: var(--model-tab-height) !important;
+}
+:where(.responsive#modeler_dialog .pentaho-tabBar) .gwt-TabBarFirst {
+  display: none;
+}
+
+:where(.responsive#modeler_dialog) .gwt-TabPanelBottom {
+  height: calc(var(--panel-height) - var(--model-tab-height)) !important;
+}
```

Comment:
Could this not be written generically for the `GwtTabBar` component?

Reply:
Excluding the last rule, the others can be easily ported to `GwtTabBar` 👍🏼